dsgrid/dataset/dataset_schema_handler_base.py

```diff
@@ -136,30 +135,30 @@ def get_columns_for_unique_arrays(self, time_dim, load_data_df):
 
         return groupby_cols
 
+    @track_timing(timer_stats_collector)
     def _check_dataset_time_consistency(self, load_data_df):
         """Check dataset time consistency such that:
         1. time range(s) match time config record;
         2. all dimension combinations return the same set of time range(s).
 
         """
         if os.environ.get("__DSGRID_SKIP_DATASET_TIME_CONSISTENCY_CHECKS__"):
+            logger.info("Skip dataset time consistency checks.")
             return
 
+        logger.info("Check dataset time consistency.")
         time_dim = self._config.get_dimension(DimensionType.TIME)
         time_dim.check_dataset_time_consistency(load_data_df)
         if time_dim.model.time_type != TimeDimensionType.NOOP:
             self._check_dataset_time_consistency_by_dimensions(time_dim, load_data_df)
 
+    @track_timing(timer_stats_collector)
     def _check_dataset_time_consistency_by_dimensions(self, time_dim, load_data_df):
         """
         Check dataset time consistency such that all dimension combinations return the same set of time ranges.
         """
-        time_ranges = time_dim.get_time_ranges()
-        assert len(time_ranges) == 1, len(time_ranges)
-        time_range = time_ranges[0]
-        # TODO: need to support validation of multiple time ranges: DSGRID-173
-        expected_timestamps = time_range.list_time_range()
-        expected_count = len(expected_timestamps)
+        logger.info("Check dataset time consistency by dimensions.")
+        expected_count = len(time_dim.list_expected_dataset_timestamps())
```

Comment:
If this is slow, we can calculate expected_count from the time range info: (start-end)/freq+1

Reply:
This one is fast.